include/deal.II/grid/grid_generator.h

```diff
@@ -873,6 +873,45 @@ namespace GridGenerator
           const double        right    = 1.,
           const bool          colorize = false);
 
+  /**
+   * Initialize the given triangulation in 2D or 3D with a generalized
+   * subdivided_hyper_L.
+   *
+   * This function produces a subdivided_hyper_rectangle with dimensions given
+   * by @p bottom_left and @p top_right, with the given number of
+   * subdivisions in each direction given in the vector @p repetitions,
+   * and with a number of cells removed, given in the vector @p num_cells_to_cut.
+   * Note that @p num_cells_to_cut contains integers, meaning that its entries
+   * can be both positive and negative. A positive number denotes
+   * cutting away cells in the 'positive' orientation, for example
+   * left to right in the x-direction, bottom to top in
+   * the y-direction, and front to back in the z-direction. A negative number
+   * denotes cutting away cells in the reverse direction, so right to left,
+   * top to bottom, and back to front.
+   *
+   * This function may be used to generate a mesh for a backward
+   * facing step, a useful domain for benchmark problems in fluid dynamics.
+   * The first image is a backward facing step in 3D, generated by
+   * removing all cells in the z-direction, and 2 cells in the
+   * positive x- and y-directions:
+   * @image html subdivided_hyper_L_3d.png
+   * And in 2D, we can cut away 1 cell in the negative x-direction, and 2 cells
+   * in the negative y-direction:
+   * @image html subdivided_hyper_L_2d.png
+   *
+   * @note This function is declared to exist for triangulations of all space
+   * dimensions, but throws an error if called in 1D.
+   *
+   * @author Mae Markowski, 2019
+   */
+  template <int dim, int spacedim>
+  void
+  subdivided_hyper_L(Triangulation<dim, spacedim> &   tria,
+                     const Point<dim> &               bottom_left,
+                     const Point<dim> &               top_right,
+                     const std::vector<unsigned int> &repetitions,
```

Comment:
I just realized that the parameter order of this function is different to all other `GridGenerator::subdivided_*` functions: They all start with `repetitions` followed by the start and end points, while this function does it the other way around.

Do you mind changing the order of the function parameters? It is up to you where you want to place `num_cells_to_cut`, though, but I would suggest to keep it as the last parameter. For this, you would need to update the `.cc`, `.inst.in` files and the test as well.

Reply:
changed the test, source, header, and inst files. good catch--thanks for the suggestion